sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -255,6 +255,18 @@ abstract class ParquetQuerySuite extends QueryTest with ParquetTest with SharedS
     }
   }
 
+  test("write and read TimestampNTZ with legacy rebase mode") {
```

Comment:
If you don't mind, please add a test prefix, `SPARK-46466: ` here for easy tracing this correctness bug.